pkg/asset/rhcos/image.go

```diff
@@ -130,31 +136,48 @@ func osImage(ctx context.Context, config *types.InstallConfig, nodeArch types.Ar
 		if a, ok := streamArch.Artifacts["openstack"]; ok {
 			return rhcos.FindArtifactURL(a)
 		}
-		return "", fmt.Errorf("%s: No openstack build found", st.FormatPrefix(archName))
+		return "", fmt.Errorf("%s: No openstack build found", streamArchPrefix)
 	case azure.Name:
 		ext := streamArch.RHELCoreOSExtensions
-		if config.Platform.Azure.CloudName == azure.StackCloud {
-			return config.Platform.Azure.ClusterOSImage, nil
+		if platform.Azure.CloudName == azure.StackCloud {
+			return platform.Azure.ClusterOSImage, nil
 		}
 		if ext == nil {
-			return "", fmt.Errorf("%s: No azure build found", st.FormatPrefix(archName))
+			return "", fmt.Errorf("%s: No extensions found in stream", streamArchPrefix)
 		}
-		azd := ext.AzureDisk
-		if azd == nil {
-			return "", fmt.Errorf("%s: No azure build found", st.FormatPrefix(archName))
+		if ext.AzureDisk == nil {
+			return "", fmt.Errorf("%s: No azure build found", streamArchPrefix)
 		}
-		return azd.URL, nil
+		azi := ext.AzureDisk.URL
+		azMP := machinePool.Platform.Azure
+		confidentialVM := azMP != nil && azMP.Settings != nil && azMP.Settings.SecurityType != ""
```

Comment:
Seen from the error in test job [periodic-ci-openshift-openshift-tests-private-release-4.21-amd64-nightly-azure-ipi-confidentialvm-vmgueststateonly-f28-destructive](https://prow.ci.openshift.org/view/gs/test-platform-results/logs/openshift-installer-10076-amd64-nightly-azure-ipi-confidentialvm-vmgueststateonly-f28-destructive/1991391050454798336) , it still tries to use Marketplace image instead of managed image when setting ConfidentialVM in `defaultMachinePlatform`.
Seems it only checks the specific machine pool's settings, but not the defaultMachinePlatform settings.